openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.cpp

```diff
@@ -1,144 +1,210 @@
 #include "openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.hpp"
 
-#include "openlr/openlr_simple_parser.hpp"
+#include "openlr/openlr_model_xml.hpp"
 
 #include "indexer/index.hpp"
 #include "indexer/scales.hpp"
 
-#include "base/stl_add.hpp"
+#include "3party/pugixml/src/utils.hpp"
 
-#include "3party/pugixml/src/pugixml.hpp"
+#include <QItemSelection>
+#include <QMessageBox>
 
-#include <QtCore/QItemSelection>
+#include <tuple>
 
+namespace
+{
+std::vector<m2::PointD> GetPoints(openlr::LinearSegment const & segment)
+{
+  std::vector<m2::PointD> result;
+  for (auto const & lrp : segment.m_locationReference.m_points)
+    result.push_back(MercatorBounds::FromLatLon(lrp.m_latLon));
+  return result;
+}
+
+// TODO(mgsergio): Optimize this.
```

Comment:
I believe I didn't like the remove part. I guess I was thinking about passing |toBeRemoved| in GetReachablePoints not to include it in the first place. No I think it's ok. So I can just remove the TODO))